### PR TITLE
Implements the PlaylistMembers and PlaylistMember Vue components for removing Recordings from Playlists

### LINKED ISSUE #2080: Playlists have tracks
* Tracks can be added or removed
* Tracks are ordered
* Removing a track from a playlist does not delete the FileSet
* Track titles should default to the FileSet label, but should be able to be customized to the playlist context without updating the FileSet
* Tracks should not show up in search results

Child of #2066

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Resolves #2080 and #2142 by addressing the following:

- Implements the `PlaylistMembers` and `PlaylistMember` Vue components for rendering Recordings in Playlists and permitting users to remove them
- Ensuring that users cannot add the same Recording more than once using the `DocumentAdder` component
- Extending `CreateProxyFileSets` in order to handle cases where ProxyFileSets are removed as members

### CODE REVIEW

File: app/change_set_persisters/change_set_persister/create_proxy_file_sets.rb [hunk 1]
```diff
@@ -2,18 +2,46 @@
...
+      else
+        return if file_set_ids.nil?
+        change_set.member_ids += proxy_file_set_ids
+      end
+
       change_set.sync
     end
 
+    def detached_proxy_file_set_ids
+      change_set.try(:detached_member_ids)
```
Comment (tpendragon) [inline app/change_set_persisters/change_set_persister/create_proxy_file_sets.rb]: Why is this all wrapped up in the same hook? Also, why does it need to be built into this at all - can't we just use the same delete logic we've always used for members?

File: app/change_sets/playlist_change_set.rb [hunk 2]
```diff
@@ -12,6 +12,7 @@ class PlaylistChangeSet < ChangeSet
   property :read_groups, multiple: true, required: false
 
   property :file_set_ids, virtual: true, type: Valkyrie::Types::Array.of(Valkyrie::Types::ID)
+  property :detached_member_ids, virtual: true, type: Valkyrie::Types::Array.of(Valkyrie::Types::ID)
```
Comment (tpendragon) [inline app/change_sets/playlist_change_set.rb]: What's the purpose of this?

File: app/decorators/playlist_decorator.rb [hunk 3]
```diff
@@ -23,4 +24,12 @@ def titles
     Array.wrap(label)
   end
   alias title titles
+
+  def proxies
```
Comment (tpendragon) [inline app/decorators/playlist_decorator.rb]: If we need this, should be called `decorated_proxies`

File: app/javascript/components/document_adder.vue [hunk 4]
```diff
@@ -53,14 +56,22 @@ export default {
       if (recording.file_set_ids.length < 1) {
         return
       }
+	// Ensure that the new FileSet IDs are unique
+	const new_file_set_ids = recording.file_set_ids.filter( file_set_id => !this.member_file_set_ids.includes(file_set_id) )
```
Comment (tpendragon) [inline app/javascript/components/document_adder.vue]: Rather than handling this here, can we just make the controller action not add them twice? Could even be in another PR.

File: app/javascript/components/playlist_members.vue [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+		return file_set_id.id
+	    })
+	},
+	submit() {},
+	buildFormData() {
+	    let form = new FormData
+	    form.append('_method', 'patch')
+
+	    for(var id of this.detached_member_ids) {
+		form.append('playlist[detached_member_ids][]', id)
```
Comment (tpendragon) [inline app/javascript/components/playlist_members.vue]: Seems like we could just send a DELETE for that ProxyFileSet and everything would clean up.

File: app/change_set_persisters/change_set_persister/create_proxy_file_sets.rb [hunk 6]
```diff
@@ -2,6 +2,8 @@
 class ChangeSetPersister
   class CreateProxyFileSets
     attr_reader :change_set_persister, :change_set
+    delegate :query_service, to: :change_set_persister
```
Comment (hackmastera) [inline app/change_set_persisters/change_set_persister/create_proxy_file_sets.rb]: there is also a `def query_service` so one of the two is redundant.

File: app/change_set_persisters/change_set_persister/create_proxy_file_sets.rb [hunk 7]
```diff
@@ -2,6 +2,8 @@
...
+    delegate :resource, to: :change_set
```
Comment (hackmastera) [inline app/change_set_persisters/change_set_persister/create_proxy_file_sets.rb]: this doesn't look like it's used any longer.

File: app/controllers/playlists_controller.rb [hunk 8]
```diff
@@ -8,7 +8,13 @@ class PlaylistsController < BaseResourceController
   )
 
   def resource_params
-    return super unless params[:recording_id]
+    values = super
+    if !values.nil? && values[:file_set_ids]
+      file_set_ids = values[:file_set_ids]
+      values[:file_set_ids] = file_set_ids.uniq
```
Comment (hackmastera) [inline app/controllers/playlists_controller.rb]: can you just add a comment saying why we get duplicates?

File: app/controllers/proxy_file_sets_controller.rb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+class ProxyFileSetsController < BaseResourceController
```
Comment (hackmastera) [inline app/controllers/proxy_file_sets_controller.rb]: why did we need a controller? oh to power the vue stuff?

**Comment by jrgriffiniii** [comment]
figgy_issues_2142_screenshot_0
